server/src/main/java/io/deephaven/server/console/completer/JavaAutoCompleteObserver.java

```diff
@@ -93,58 +82,74 @@ public void onNext(AutoCompleteRequest value) {
+                }
+                case GET_SIGNATURE_HELP: {
+                    // Not implemented. Return empty signature list
+                    response.setSignatures(GetSignatureHelpResponse.newBuilder().build());
+                    break;
+                }
+                case GET_HOVER: {
+                    // Not implemented. Return empty hover help
+                    response.setHover(
+                            GetHoverResponse.newBuilder().setContents(MarkupContent.newBuilder().build()).build());
+                    break;
+                }
+                case GET_DIAGNOSTIC: {
+                    // Not implemented. Return empty diagnostics
+                    response.setDiagnostic(GetPullDiagnosticResponse.newBuilder().build());
+                    break;
                 }
-                safelyOnNext(responseObserver,
-                        AutoCompleteResponse.newBuilder()
-                                .setCompletionItems(GetCompletionItemsResponse.newBuilder()
-                                        .setSuccess(false)
-                                        .setRequestId(request.getRequestId()))
-                                .build());
-                return;
             }
 
-            int offset = LspTools.getOffsetFromPosition(parsed.getSource(),
-                    request.getPosition());
-            final Collection<CompletionItem.Builder> results =
-                    completer.runCompletion(parsed, request.getPosition(), offset);
-            final GetCompletionItemsResponse mangledResults =
-                    GetCompletionItemsResponse.newBuilder()
+            safelyOnNext(responseObserver,
+                    response
                             .setSuccess(true)
-                            .setRequestId(request.getRequestId())
-                            .addAllItems(results.stream().map(
-                                    // insertTextFormat is a default we used to set in constructor; for now, we'll
-                                    // just process the objects before sending back to client
-                                    item -> item.setInsertTextFormat(2).build())
-                                    .collect(Collectors.toSet()))
-                            .build();
+                            .setRequestId(requestId)
+                            .build());
 
+        } catch (CompletionCancelled exception) {
+            if (log.isTraceEnabled()) {
+                log.trace().append("Completion canceled").append(exception).endl();
+            }
             safelyOnNext(responseObserver,
                     AutoCompleteResponse.newBuilder()
-                            .setCompletionItems(mangledResults)
+                            .setSuccess(false)
+                            .setRequestId(request.getRequestId())
                             .build());
-        } catch (Exception exception) {
+        } catch (Throwable exception) {
```

Comment:
That's just what was in `getCompletionItems` error handling. I think the only difference is rethrowing the exception if it's an Error which is from `PythonAutoCompleteObserver`.

If the parser fails or some other step in `getCompletionItems` fails, would it be an `Error` or just a `Throwable`?

Reply:
It would be an Exception, unless it was something terrible like "out of memory" or "i actually cant talk to python at all" or "this thread is interrupted because the server is shutting down" - things we can't reasonably recover from.

Exception and Error both extend from Throwable, if we have to notice exiting a block in error you would catch throwable, but if you are just handling reasonable failures, it should be an exception